pkg/infrastructure/azure/azure.go

```diff
@@ -227,17 +228,19 @@ func (p *Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput
 	logrus.Debugf("BlobContainer.ID=%s", *blobContainer.ID)
 
 	// Upload the image to the container
-	_, err = CreatePageBlob(ctx, &CreatePageBlobInput{
-		StorageURL:         storageURL,
-		BlobURL:            blobURL,
-		ImageURL:           imageURL,
-		ImageLength:        imageLength,
-		StorageAccountName: storageAccountName,
-		StorageAccountKeys: storageAccountKeys,
-		CloudConfiguration: cloudConfiguration,
-	})
-	if err != nil {
-		return err
+	if _, ok := os.LookupEnv("OPENSHIFT_INSTALL_SKIP_IMAGE_UPLOAD"); !ok {
+		_, err = CreatePageBlob(ctx, &CreatePageBlobInput{
+			StorageURL:         storageURL,
+			BlobURL:            blobURL,
+			ImageURL:           imageURL,
+			ImageLength:        imageLength,
+			StorageAccountName: storageAccountName,
+			StorageAccountKeys: storageAccountKeys,
+			CloudConfiguration: cloudConfiguration,
+		})
+		if err != nil {
+			return err
+		}
```

Comment:
need to extend this so you also skip the gallery image and gallery image version

Reply:
Maybe you can move the image/gallery related calls to a function and either:
```go
if _, ok := os.LookupEnv("OPENSHIFT_INSTALL_SKIP_IMAGE_UPLOAD"); !ok {
    if err := createVMImage(ctx, createImageInput{...}); err != nil {
        return err
    }
}
```
Or always call it but check the env var inside `createVMImage` and return immediately if it's set.

Reply:
I feel like doing this is more easier than passing a bunch of inputs to another function just for an extra if condition.